### ISSUE TITLE
Should spatial vectors be required to have geometries?

### ISSUE THREAD

**Issue by phargogh** [natcap/invest#64]
InVEST has, for quite some time, assumed that if we have a spatial vector layer*, all features in the layer must have geometries.  In some, but not all, cases, these geometries must also be valid.

Before just jumping into attempting to solve this, it would be helpful to see how we handle these cases throughout InVEST and pygeoprocessing.  Are there examples where geometries are not required to be present?  And is this something that is reasonable to specify in `ARGS_SPEC` and validate in `validation`?

* Note that, especially with GeoPackages, it's [possible to create an aspatial layer](https://gist.github.com/phargogh/f654a8de747544272f356fb34024dc87#file-aspatial-py).

**Comment by phargogh** [comment]
This might be similar to the undefined nodata issue: if it's supported, we should handle it.

**Comment by davemfish** [comment]
> This might be similar to the undefined nodata issue: if it's supported, we should handle it.

I tend to agree with this. Skipping & warning about missing geometries is a better user-experience than crashing.
